projectroles/app_settings.py

```diff
@@ -101,37 +107,22 @@ class AppSettingAPI:
     @classmethod
     def _check_project_and_user(cls, scope, project, user):
         """
-        Ensure one of the project and user parameters is set.
+        Ensure project and user parameters are set according to scope.
 
-        :param scope: Scope of Setting (USER, PROJECT, PROJECT_USER)
+        :param scope: Scope of Setting (USER, PROJECT, PROJECT_USER, SITE)
         :param project: Project object
         :param user: User object
         :raise: ValueError if none or both objects exist
         """
-        if scope == APP_SETTING_SCOPE_PROJECT:
-            if not project:
-                raise ValueError('Project unset for setting with project scope')
-            if user:
-                raise ValueError('User set for setting with project scope')
-        elif scope == APP_SETTING_SCOPE_USER:
-            if project:
-                raise ValueError('Project set for setting with user scope')
-            if not user:
-                raise ValueError('User unset for setting with user scope')
-        elif scope == APP_SETTING_SCOPE_PROJECT_USER:
-            if not project:
-                raise ValueError(
-                    'Project unset for setting with project_user scope'
-                )
-            if not user:
-                raise ValueError(
-                    'User unset for setting with project_user scope'
+        if not APP_SETTING_SCOPE_ARGS[scope] == {
+            'project': project is not None,
+            'user': user is not None,
+        }:
+            raise ValueError(
+                'Project and/or user are set incorrect for setting with {} scope'.format(
```

Comment:
I believe this row goes beyond 80 characters in length :) (black doesn't catch ones with strings)